### PR TITLE
Add core to cpp

### PR DESCRIPTION
**Pull request by tanaya-mankad**
This branch is an incremental step in making the CPP-generation code generic for lattice (removing 205-specific source files, adding lattice-specific ones).

### CODE REVIEW

File: lattice/cpp/templates/enum-info.h.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#ifndef TYPEINFO_205_H_
+#define TYPEINFO_205_H_
```
Comment (nealkruis) [inline lattice/cpp/templates/enum-info.h.txt:2]: Generalize

File: lattice/cpp/templates/enum-info.h.txt
Comment (nealkruis) [inline lattice/cpp/templates/enum-info.h.txt:1]: Jinja template file extensions should be `.j2`, not `.txt`.

File: lattice/cpp/templates/load-object.h.txt
Comment (nealkruis) [inline lattice/cpp/templates/load-object.h.txt:1]: See other template comments.

File: lattice/header_entries.py [hunk 2]
```diff
@@ -611,6 +614,36 @@ def translate(self, input_file_path, container_class_name, schema_base_class_nam
                 # objects might be included and used from elsewhere.
                 ObjectSerializationDeclaration(base_level_tag, self._namespace)
 
+    # .............................................................................................
```
Comment (nealkruis) [inline lattice/header_entries.py:617]: This really should use the new Schema class!

File: lattice/util.py [hunk 3]
```diff
@@ -4,3 +4,6 @@ def snake_style(s):
     #return ''.join(['_'+c.lower() if c.isupper() else c for c in s]).lstrip('_')
     a = re.compile('((?<=[a-z0-9])[A-Z]|(?!^)[A-Z](?=[a-z]))')
     return a.sub(r'_\1', s).lower()
+
+def bigladder_filename_style(s):
```
Comment (nealkruis) [inline lattice/util.py:8]: This function name is more specific than its functionality. Maybe this should be called "hyphen_separated_lowercase"?

**Review by nealkruis — CHANGES_REQUESTED** [review]

File: lattice/header_entries.py (hunk 2 above)
Comment (tanaya-mankad) [reply lattice/header_entries.py:617]: Not yet. Next PR.

File: lattice/cpp/templates/enum-info.h.txt
Comment (tanaya-mankad) [reply lattice/cpp/templates/enum-info.h.txt:1]: OK, and how about hyphen vs underscore? The file is a j2 type, so should it stick with the naming convention of its target, or its source?

Comment (nealkruis) [reply lattice/cpp/templates/enum-info.h.txt:1]: It should use the same convention as its target (i.e., hyphens).

Comment (tanaya-mankad) [reply lattice/cpp/templates/enum-info.h.txt:1]: Cool, that's what I checked in.

**Pull request merged**